coercion.ts

```diff
@@ -113,121 +110,68 @@ function coerceFile(file: unknown) {
-          ...PipeItem<unknown, unknown, BaseIssue<unknown>>[],
-        ]
-      >,
->(
-  originalSchema:
-    | T
-    | (T extends GenericSchema
-        ? SchemaWithPipe<
-            [T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]
-          >
-        : SchemaWithPipeAsync<
-            [T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]
-          >),
-  coercionSchema: E,
-) {
-  if ("pipe" in originalSchema) {
-    if (originalSchema.async && coercionSchema.async) {
-      return pipeAsync(coercionSchema, ...originalSchema.pipe.slice(1));
-    }
-    // @ts-expect-error
-    return pipe(coercionSchema, ...originalSchema.pipe.slice(1));
-  }
-
-  return coercionSchema;
-}
-
 /**
  * Generate a wrapped schema with coercion
  * @param type The schema to be coerced
- * @param originalSchema The original schema
- * @param schemaType The schema type
+ * @param rewrap Whether the result schema should be rewrapped with the `type` schema.
+ *   See <https://github.com/chimame/conform-to-valibot/issues/53> for cases that need rewrapping.
  * @returns The coerced schema
  */
 function generateWrappedSchema<T extends GenericSchema | GenericSchemaAsync>(
   type: T,
-  originalSchema: T,
-  schemaType?: "nullish" | "optional",
+  rewrap = false,
 ) {
-  // @ts-expect-error
-  const { coerced, schema: wrapSchema } = enableTypeCoercion(type.wrapped);
+  const { transformAction, schema: wrapSchema } = enableTypeCoercion(
+    // @ts-expect-error
+    type.wrapped,
+  );
 
-  if (coerced) {
+  if (transformAction) {
     // `expects` is required to generate error messages for `TupleSchema`, so it is passed to `UnkonwSchema` for coercion.
     const unknown = { ...valibotUnknown(), expects: type.expects };
-    if (type.async) {
-      switch (schemaType) {
-        case "nullish":
-          return {
-            coerced: false,
-            schema: nullishAsync(pipeAsync(unknown, wrapSchema.pipe[1], type)),
-          };
-        case "optional":
-          return {
-            coerced: false,
-            schema: optionalAsync(pipeAsync(unknown, wrapSchema.pipe[1], type)),
-          };
-        default:
-          return {
-            coerced,
-            schema: pipeAsync(unknown, wrapSchema.pipe[1], type),
-          };
-      }
-    }
-    switch (schemaType) {
-      case "nullish":
-        return {
-          coerced: false,
-          schema: nullish(pipe(unknown, wrapSchema.pipe[1], type)),
-        };
-      case "optional":
-        return {
-          coerced: false,
-          schema: optional(pipe(unknown, wrapSchema.pipe[1], type)),
-        };
-      default:
-        return {
-          coerced,
-          schema: pipe(unknown, wrapSchema.pipe[1], type),
-        };
+    const schema = type.async
+      ? pipeAsync(unknown, transformAction, type)
+      : pipe(unknown, transformAction, type);
+
+    if (rewrap) {
+      return {
+        transformAction: undefined,
+        schema: type.reference(schema),
```

Comment:
The original schema information can be obtained from the `reference` property, so by using the schema defined in the `reference` property, the import declaration can be removed.
This is a great change that may reduce the bundle size when using this library. 👍 

However, I am a bit concerned about which version of valibot this `reference` property was added to. I don't think it was in `0.32.0`, the minimum version supported by `conform-to-valibot`.

I accept this change itself, so I think it would be better to increase the minimum version of valibot that is supported.